rdkit/Chem/Pharm3D/EmbedLib.py

```diff
@@ -1259,6 +1364,309 @@ def ComputeChiralVolume(mol, centerIdx, confId=-1):
+    mappings = self.mappings
+    del self.mappings
+
+    if len(mappings) == 0:
+      raise EmbedException(ExitState.NO_MAPPINGS_FOUND)
+    return mappings
+
+  def get_ph4_constraints_from_mapping(self, mapping, pharmacophore):
+    """
+    NOTE: At least 2018.06.01 rdkit version must be used, otherwise several
+    bounds smoothing errors will occur here.
+    """
+    if not self.settings.usePh4Constraints:
+      return [(None, mapping, self.pharmacophore)]
+
+    ph4_constraints = []
+    self.ligand = Chem.AddHs(self.ligand)
+    # recommended for good 3D conformations
+    Chem.AssignStereochemistry(self.ligand, cleanIt=True, force=True)
+
+    try:
+      _, embeddings, _ = EmbedPharmacophore(
+        self.ligand, mapping, pharmacophore, self.settings.maxEmbeddings)
+    except ValueError as e:
+      return []    # This is a bogus mapping
+    else:
+      if len(embeddings) == 0:
+        return []  # This is a bogus mapping
+
+    coords = [embed.GetConformer(0).GetPositions() for embed in embeddings]
+    for coordTuple in coords:
+      ph4_constraints.append(coordTuple)
+    return ph4_constraints
+
+  def get_conformers_from_ph4_constraint(self, ph4_constraint, mapping, pharmacophore):
+    if self.settings.usePh4Constraints:
+      embedding = self._embed_conformers_constrained(ph4_constraint, mapping)
+    else:
+      embedding = self._embed_conformers_unconstrained()
+    if embedding is None:
+      return None
+    return GetAlignedConformers(embedding, pharmacophore, mapping, self.settings.minimize)
+
+  def _embed_conformers_unconstrained(self):
+    self.ligand.RemoveAllConformers()
+    numConfs = min(GetNumOptimalConformers(self.ligand), self.settings.maxConformers)
+    mol = Chem.Mol(self.ligand)
+
+    if self.settings.useETKDG:
+      # FIX: EmbedMultipleConfs args != EmbedParameters fields
+      rdDistGeom.EmbedMultipleConfs(mol, numConfs, rdDistGeom.ETKDG())
+    else:
+      rdDistGeom.EmbedMultipleConfs(mol, numConfs)
+    if mol.GetNumConformers() > 0:
+      return mol
+
+  def _embed_conformers_constrained(self, ph4_constraint, mapping):
+    self.ligand.RemoveAllConformers()
+    mol = Chem.Mol(self.ligand)
+    numConfs = max(1,
+                   min(GetNumOptimalConformers(self.ligand),
+                       self.settings.maxConformers)
+                   // self.settings.maxMappings
+                   // self.settings.maxEmbeddings) # enforce floor int division
+
+    # Currently we select the first atom from each MolFeature. This can
+    # (and should) be changed.
+    staticAts = [molFeat[0] for molFeat in mapping]
+    coordMap = {i:Geometry.Point3D(*ph4_constraint[i]) for i in range(
+          len(mapping)) if i in staticAts}
+    assert len(coordMap) > 0
+
+    if self.settings.useETKDG:
+      # params = rdDistGeom.ETKDG()
+      # params.coordMap = coordMap  # for now, need to manually specify coordMap
+      rdDistGeom.EmbedMultipleConfs(
```

Comment:
Why not just use keyword arguments here to indicate which values you are intentionally changing from the default?
or, for that matter, you could also use a parameters object. From the fact you have that commented out it looks like you might have encountered a problem with that method. What was it?